File: C++/1_Bubble Sort/Bubble Sort.cpp
#include <iostream>
using namespace std;

void swap(int *a, int *b){
	int temp = *a;
	*a = *b;
	*b = temp;
}

void bubbleSort(int arr[], int n){
	bool swapped;

	for(int i = 0; i < n-1; i++){
		swapped = false;
		for(int j = 0; j < n-i-1; j++){
			if(arr[j] > arr[j+1]){
				swap(&arr[j], &arr[j+1]);
				swapped = true;
			}
		}

		if(swapped == false){
			break;
		}
	}
}

void printArray(int arr[], int n){
	for(int i = 0; i < n; i++){
		cout<<arr[i]<<" ";
	}
}

int main() {
	int arr[] = {5, 2, 42, 6, 1, 3, 2};
	int n = sizeof(arr) / sizeof(arr[0]);
	bubbleSort(arr, n);
	printArray(arr, n);
	return 0;
}

File: Ruby/1_Bubble Sort/Bubble Sort.rb
def bubbleSort(arr)
	swapped = true
	for i in 0...(arr.length - 1)
		swapped = false
		for j in 0...(arr.length - i - 1)
			if arr[j] > arr[j+1]
				arr[j], arr[j+1] = arr[j+1], arr[j]
				swapped = true
			end
		end
		break if swapped == false
	end
end

nums = [3, 2, 1, 5, 21, 8, 6, 9, 4, 7]
bubbleSort(nums)
p nums